src/components/profile/ProfileDetails.jsx

```diff
@@ -1,11 +1,14 @@
-import React from "react";
+import React, { useEffect } from "react";
 import { useAuth } from "../../contexts";
 import { useNavigate } from "react-router-dom";
 import "./profile.css";
 
 function ProfileDetails() {
   const { setAuthToken, authUser, setAuthUser } = useAuth();
   const navigate = useNavigate();
+  useEffect(() => {
+    if (localStorage.authUser) setAuthUser(JSON.parse(localStorage.authUser));
+  }, []);
   function logoutHandler() {
     localStorage.removeItem("authToken");
     localStorage.removeItem("authUser");
```

Comment:
not sure but directly storing user data to local storage ...will it not concern privacy issues